### PR TITLE
[Merged by Bors] - refactor(Measure): improve defeq for `⊤`

### PR DESCRIPTION
**Pull request by urkud**
Now `(⊤ : Measure α).toOuterMeasure = ⊤` is defeq.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean [hunk 1]
```diff
@@ -1057,35 +1057,36 @@ instance instCompleteSemilatticeInf [MeasurableSpace α] : CompleteSemilatticeIn
 #align measure_theory.measure.complete_semilattice_Inf MeasureTheory.Measure.instCompleteSemilatticeInf
 
 instance instCompleteLattice [MeasurableSpace α] : CompleteLattice (Measure α) :=
-  { /- Porting note:
-    Adding an explicit `top` made `leanchecker` fail in Lean3 because of lean#364,
-    but in Lean4 it's all right.
-    top := (⊤ : OuterMeasure α).toMeasure
-      (by rw [OuterMeasure.top_caratheodory]; exact le_top)
-    le_top := fun a s hs => by
-      rcases s.eq_empty_or_nonempty with rfl | h <;>
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Measure/MeasureSpace.lean:1066]: Any reason not to use this lemma?

bors d+

Comment (urkud) [reply Mathlib/MeasureTheory/Measure/MeasureSpace.lean:1066]: This way I have defeq for `toOuterMeasure` projection.

**Comment by urkud** [comment]
bors merge

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/MeasureTheory/Measure/MeasureSpace.lean:1066]: I meant `s.eq_empty_or_nonempty`

**Pull request closed without merging**